src/metaswitch/crest/api/base.py

```diff
@@ -188,25 +188,45 @@ def update_latency(self, latency):
                                  (accepted_percent, err, hss_overloads, self.bucket.rate, new_rate))
                 self.bucket.update_rate(new_rate)
             elif err < self.INCREASE_THRESHOLD:
-                # latency is sufficiently below the target, so we can increase by an additive
-                # factor - weighted by how far below target we are.
-                new_rate = self.bucket.rate + (-err) * self.bucket.max_size * self.INCREASE_FACTOR
-                _log.info("Accepted %f%% of requests, latency error = %f, increase rate %f to %f" %
-                                (accepted_percent, err, self.bucket.rate, new_rate))
-                self.bucket.update_rate(new_rate)
+                # latency is sufficiently below the target, so increasing the permitted
+                # request rate would be sensible; but first check that we are using a
+                # significant proportion of the current rate - if we're allowing 100
+                # requests/sec, and we get 1 request/sec during a quiet period, we will
+                # handle that well, but it is not sufficient evidence that we can increase the rate.
+                max_permitted_requests = self.bucket.rate * seconds_since_last_update
+
+                # Arbitrary threshold of at least 50% of maximum permitted requests
+                minimum_threshold = max_permitted_requests * 0.5
+
+                if (self.accepted > minimum_threshold):
+                    new_rate = self.bucket.rate + (-err) * self.bucket.max_size * self.INCREASE_FACTOR
+                    _log.info("Accepted %.2f%% of requests, latency error = %f, increase rate %f to %f"
+                              " based on %d accepted requests in last %.2f seconds" %
+                              (accepted_percent, err, self.bucket.rate, new_rate,
+                               self.accepted, seconds_since_last_update))
+                    self.bucket.update_rate(new_rate)
+                else:
+                    _log.info("Only handled %d requests in the last %.2f seconds, rate remains unchanged."
+                              " Minimum threshold for change is %f" %
+                              (self.accepted, seconds_since_last_update, minimum_threshold))
             else:
                 _log.info("Accepted %f%% of requests, latency error = %f, rate %f unchanged" %
                                 (accepted_percent, err, self.bucket.rate))
 
+            self.accepted = 0
+            self.rejected = 0
+            self.adjust_count = self.REQUESTS_BEFORE_ADJUSTMENT
+            self.last_adjustment_time = monotonic()
+
         penaltycounter.reset_hss_penalty_count()
 
 # Create load monitor with target latency of 100ms, maximum bucket size of
-# 20 requests, initial and minimum token rate of 10 per second
-loadmonitor = LoadMonitor(0.1, 20, 10, 10)
+# 100 requests, initial and minimum token rate of 10 per second
+loadmonitor = LoadMonitor(0.1, 100, 100, 10)
```

Comment:
A nicer solution here might be to make this configurable, but I don't think it's worth it now.
Do these values match the cpp defaults?